### PR TITLE
add dockerfile, update readme and include deploy script

### LINKED ISSUE #10: Docker Image for Script
So to avoid issues with strict dependency on python version

### PR DESCRIPTION
**Pull request by bh2smith**
1. Updates readme with changes that @fedgiac had previously made. 
2. Adds docker file
3. Adds docker image deployment configuration. 

closes #10

### CODE REVIEW

**Review by josojo — APPROVED** [review]

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bh2smith) [inline .github/workflows/deploy.yaml:22]: So @nlordell  - I don't need to do anything here, this secret will be available and if I tag the branch an image will be pushed to docker hub?

Comment (nlordell) [reply .github/workflows/deploy.yaml:22]: From the [docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret):

> At the start of each workflow run, GitHub automatically creates a unique `GITHUB_TOKEN` secret to use in your workflow. You can use the `GITHUB_TOKEN` to authenticate in a workflow run.

File: README.md [hunk 2]
```diff
@@ -81,29 +78,44 @@ However, many IDEs can be configured to auto format on save.
...
 
+This project has a strict requirement of python >= 3.10 which may not be common version for most operating systems. 
+In order to make the reimbursement effort seamless, we have prepared the following docker container.
 
+```shell
+docker pull ghcr.io/cowprotocol/solver-rewards
+# Prepare environment variables
+cp .env.sample .env  # Fill in your Dune Credentials
+# Run
+docker run -it --rm --env-file .env -v $(PWD):/app/out solver-rewards --start 'YYYY-MM-DD'
```
Comment (nlordell) [inline README.md:118]:
```suggestion
docker run -it --rm --env-file .env -v $(PWD):/app/out ghcr.io/cowprotocol/solver-rewards --start 'YYYY-MM-DD'
```

**Review by nlordell — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bh2smith) [reply README.md:118]: Whoops. I forgot to include this! Will make a follow up.

Comment (bh2smith) [reply README.md:118]: Fixed in #12

**Pull request merged**